futures-util/src/io/mod.rs

```diff
@@ -99,6 +192,34 @@ pub trait AsyncReadExt: AsyncRead {
     ///
     /// The two halves returned implement the `Read` and `Write` traits,
     /// respectively.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::io::AsyncReadExt;
+    /// use std::io::Cursor;
+    ///
+    /// let mut reader = Cursor::new([1, 2, 3, 4]);
+    /// let mut buffer = [0, 0, 0, 0, 4, 3, 2, 1];
```

Comment:
One final tiny thing: Here I'd prefer to not reuse the numbers and instead use different numbers like `5, 6, 7, 8`. This makes it more clear what's happening